gluten-iceberg/src/main/java/io/glutenproject/substrait/rel/IcebergLocalFilesNode.java

```diff
@@ -57,7 +35,79 @@ class DeleteFile {
+      Map<String, List<DeleteFile>> deleteFilesMap) {
     super(index, paths, starts, lengths, partitionColumns, fileFormat, preferredLocations);
+    this.deleteFilesMap = deleteFilesMap;
+  }
+
+  @Override
+  protected void processFileBuilder(ReadRel.LocalFiles.FileOrFiles.Builder fileBuilder) {
+    List<DeleteFile> deleteFiles =
+        deleteFilesMap.getOrDefault(fileBuilder.getUriFile(), Collections.emptyList());
+    ReadRel.LocalFiles.FileOrFiles.IcebergReadOptions.Builder icebergBuilder =
+        ReadRel.LocalFiles.FileOrFiles.IcebergReadOptions.newBuilder();
+
+    switch (fileFormat) {
+      case ParquetReadFormat:
+        ReadRel.LocalFiles.FileOrFiles.ParquetReadOptions parquetReadOptions =
+            ReadRel.LocalFiles.FileOrFiles.ParquetReadOptions.newBuilder()
+                .setEnableRowGroupMaxminIndex(
+                    GlutenConfig.getConf().enableParquetRowGroupMaxMinIndex())
+                .build();
+        icebergBuilder.setParquet(parquetReadOptions);
+        break;
+      case OrcReadFormat:
+        ReadRel.LocalFiles.FileOrFiles.OrcReadOptions orcReadOptions =
+            ReadRel.LocalFiles.FileOrFiles.OrcReadOptions.newBuilder().build();
+        icebergBuilder.setOrc(orcReadOptions);
+        break;
+      default:
+        throw new UnsupportedOperationException(
+            "Unsupported file format " + fileFormat.name() + " for iceberg data file.");
+    }
+
+    for (DeleteFile delete : deleteFiles) {
+      ReadRel.LocalFiles.FileOrFiles.IcebergReadOptions.DeleteFile.Builder deleteFileBuilder =
+          ReadRel.LocalFiles.FileOrFiles.IcebergReadOptions.DeleteFile.newBuilder();
+      ReadRel.LocalFiles.FileOrFiles.IcebergReadOptions.FileContent fileContent;
+      switch (delete.content()) {
+        case EQUALITY_DELETES:
+          fileContent =
+              ReadRel.LocalFiles.FileOrFiles.IcebergReadOptions.FileContent.EQUALITY_DELETES;
+          break;
+        case POSITION_DELETES:
+          fileContent =
+              ReadRel.LocalFiles.FileOrFiles.IcebergReadOptions.FileContent.POSITION_DELETES;
+          break;
+        default:
+          throw new UnsupportedOperationException(
+              "Unsupported FileCount " + delete.content().name() + " for delete file.");
+      }
+      deleteFileBuilder.setFileContent(fileContent);
+      deleteFileBuilder.setFilePath(delete.path().toString());
+      deleteFileBuilder.setFileSize(delete.fileSizeInBytes());
+      deleteFileBuilder.setRecordCount(delete.recordCount());
+      switch (delete.format()) {
+        case PARQUET:
+          ReadRel.LocalFiles.FileOrFiles.ParquetReadOptions parquetReadOptions =
+              ReadRel.LocalFiles.FileOrFiles.ParquetReadOptions.newBuilder()
+                  .setEnableRowGroupMaxminIndex(
+                      GlutenConfig.getConf().enableParquetRowGroupMaxMinIndex())
+                  .build();
+          deleteFileBuilder.setParquet(parquetReadOptions);
```

Comment:
Should the deletion files share same read options as the data file?

Reply:
Iceberg allows the format of delete file and data file to be different, but in most cases they are consistent.